lib/diaspora_federation/salmon/xml_payload.rb

```diff
@@ -54,6 +54,21 @@ def self.unpack(xml)
         populate_entity(klass, data)
       end
 
+      # Constructs string from XML payload to sign and verify signature for
+      #
+      # @param [Nokogiri::XML::Element] xml payload XML root node
+      # @param [[String]] exclude_accessors xml nodes of an entity to exclude (entities with
+      # "signature" in their names are excluded by default)
+      # @return [String] a string to sign or verify
+      def self.signable_string(xml, exclude_accessors=[])
+        data = xml.at_xpath("post/*[1]")
+        data.children.map { |node|
+          node.text unless node.name.match(/signature/) ||
+            exclude_accessors.include?(node.name) ||
+            node.text.strip.empty?
```

Comment:
Not sure we should skip only whitespace nodes here. Any rationale? Also extracting the whole condition into a private method seems worthwhile